### PR TITLE
[Chore] Add lint and formatter in PR comments and update xcov report

### LINKED ISSUE #11: [Chore] Add lint and formatter in PR comments  and update xcov report
## Why

Need to integrate lint in the CI so that any lint rule violation can be shown in PR comments.
 
## Acceptance Criteria?

- Add Swift lint in CI pipeline
- Add Swift Formatter
- Make sure it works properly
- Update XCOV reports according to this [comment](https://github.com/nimblehq/ci-cd-sample-ios/pull/14#discussion_r987724092)

### PR DESCRIPTION
**Pull request by Shayokh144**
https://github.com/nimblehq/ci-cd-sample-ios/issues/11

## What happened

Add lint and formatter in PR comments and update xcov report
 
## Insight

- Add Swift lint in CI pipeline with binary
- Add Swift Formatter with binary
- To Make sure it works properly add a lint violation in swift file
- Update XCOV reports according to this [comment](https://github.com/nimblehq/ci-cd-sample-ios/pull/14#discussion_r987724092)
- Post details report from xcov markdown
 
## Proof Of Work

**POW comments are in the comment section**
Screenshot 2022-10-06 at 11 35 32 AM

**POW of details coverage report**
Screenshot 2022-10-06 at 2 27 44 PM

### CODE REVIEW

File: WeatherTodayTests/Sources/Specs/Supports/Extensions/Foundation/OptionalUnwrapSpec.swift [hunk 1]
```diff
@@ -3,7 +3,7 @@ import Quick
 
 @testable import WeatherToday
 
-final class OptionalUnwrapSpec: QuickSpec {
+final class OptionalUnwrapSpec: QuickSpec{
```
Comment (Shayokh144) [inline WeatherTodayTests/Sources/Specs/Supports/Extensions/Foundation/OptionalUnwrapSpec.swift]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/ci-cd-sample-ios/blob/3df77319a1d3495943321a24797fc7d953905cbd/WeatherTodayTests/Sources/Specs/Supports/Extensions/Foundation/OptionalUnwrapSpec.swift#L6"></span>Opening braces should be preceded by a single space and on the same line as the declaration.<br />
<code>opening_brace</code> <code>OptionalUnwrapSpec.swift:6</code></td>
    </tr>
  </tbody>
</table>

**Comment by Shayokh144** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Capabilities for Signing &amp; Capabilities may not function correctly because its entitlements use a placeholder team ID. To resolve this, select a development team in the WeatherToday editor.</td>
    </tr>
  </tbody>
</table>

## Current coverage for WeatherToday.app is `14.88%`
✅ *No files affecting coverage found*

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

## Coverage Table

Files changed | - | - 
--- | --- | ---
NetworkAPIProtocol.swift | `0.00%` | :skull:
Navigator.swift | `0.00%` | :skull:
RequestConfiguration.swift | `0.00%` | :skull:
UIView+Subviews.swift | `0.00%` | :skull:
NetworkAPI.swift | `0.00%` | :skull:
Color+Application.swift | `0.00%` | :skull:
AppDelegate.swift | `100.00%` | :white_check_mark:
HomeViewController.swift | `100.00%` | :white_check_mark:
Optional+Unwrap.swift | `100.00%` | :white_check_mark:

---



<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WeatherTodayTests/Sources/Specs/Supports/Extensions/Foundation/OptionalUnwrapSpec.swift (hunk 1 above)
Comment (Shayokh144) [reply WeatherTodayTests/Sources/Specs/Supports/Extensions/Foundation/OptionalUnwrapSpec.swift]: resolved [here](https://github.com/nimblehq/ci-cd-sample-ios/pull/15/commits/9c8872293f5553c424b70f98a92a991f54e6cae2)

**Review by blyscuit — APPROVED** [review]

**Review by vnntsu — APPROVED** [review]
LGTM!

**Pull request merged**